orchestrator/main_loops.go

```diff
@@ -331,21 +331,38 @@ func (p *gravityOrchestrator) BatchRequesterLoop(ctx context.Context) (err error
 						return fmt.Errorf("failed to get Ethereum gas estimate: %w", err)
 					}
 
-					usdEthPrice, err := p.priceFeeder.QueryUSDPriceByCoinID(coingecko.EthereumCoinID)
+					usdEthPrice, err := p.oracle.GetPrice(oracle.BaseSymbolETH)
 					if err != nil {
 						return err
 					}
 
-					usdEthPriceDec = decimal.NewFromFloat(usdEthPrice)
+					usdEthPriceDec, err = decimal.NewFromString(usdEthPrice.String())
+					if err != nil {
+						return err
+					}
 
 					for _, token := range unbatchedTokensWithFees {
 						if _, ok := tokensPrices[token.Token]; !ok {
-							price, err := p.priceFeeder.QueryTokenUSDPrice(ethcmn.HexToAddress(token.Token))
+							baseSymbol, err := p.coinGecko.GetTokenSymbol(ethcmn.HexToAddress(token.Token))
 							if err != nil {
 								return err
 							}
 
-							tokensPrices[token.Token] = decimal.NewFromFloat(price)
+							price, err := p.oracle.GetPrice(baseSymbol)
+							if err != nil {
+								// maybe the token was not subscribed yet
+								if err := p.oracle.SubscribeSymbols(baseSymbol); err != nil {
```

Comment:
Why are we doing this here and not in IsBatchProfitable? I think we should do it in both places, wdyt?

Reply:
Totally agree, good catch ;D